feature/experimental/p4rt/otg_tests/traceroute_packetin_test/README.md

```diff
@@ -1,34 +1,243 @@
+
+*   Connect a P4RT client and configure the forwarding pipeline. Install P4RT table entries required for traceroute.
+	These are located in [p4rt_utils.go] (https://github.com/openconfig/featureprofiles/blob/main/internal/p4rtutils/p4rtutils.go)
+	p4rtutils.ACLWbbIngressTableEntryGet(packetIO.GetTableEntry(delete, isIPv4))
 
 
 ## Procedure
 
-*	Connect ATE port-1 to DUT port-1, ATE port-2 to DUT port-2.
-*	TODO: Install the set of routes on the device.
-*	Enable the P4RT server on the device.
-*	Connect a P4RT client and configure the forwarding pipeline. InstallP4RT table 	entries required for traceroute.
-*	Send IPv4 packets from the ATE with TTL=1 and verify that packets with TTL=1 are received by the client.
-*	Send IPv6 packets from the ATE with HopLimit=1 and verify that packets with HopLimit=1 are received by the client.
-*	Verify that the packets have both ingress_singleton_port and egress_singleton_port metadata set.
+At the start of each of the following scenarios, ensure:
+
+*   All ports are up and baseline is reset as above.
+
+Unless otherwise specified, all the tests below should use traffic with
+`dscp_encap_a_1` referenced int he VRF selection policy.
+
+### TODO: Test-1
+
+Tests that traceroute with TTL=1 for a packet that would match the VRF
+selection policy for encap has target_egress_port set based on the
+encap VRF.
+
+*   Send packets to DUT port-1 with outer packet header TTL=1. The outer v4 header has the destination
+    addresses 138.0.11.8. verify that packets with TTL=1 are received by the client.
+
+*   Verify that the punted packets have both ingress_port and target_egress_port metadata set.
+The distribution of packets should have target_egress_port set with port 2 10% of the time, port 3 30%, port 4 60%.
+
+### TODO: Test-2
+
+Tests that traceroute with TTL=1 for a packet that would match the VRF
+selection policy for default has target_egress_port set based on the
+default VRF.
+
+*   Send packets to DUT port-1 with packet TTL=1. The v4 header has the destination
+    address 203.0.113.0.  Verify that packets with TTL=1 are received by the client.
+*   Verify that the packets have both ingress_port and target_egress_port metadata set.
+target_egress_port should be dut port 8.
+
+
+### TODO: Test-3
+
+Tests that a packet punted due to TTL=1 for a packet that would
+otherwise hit a transit VRF has target_egress_port set based on that
+transit VRF.
+
+*   Send 4in4 (IP protocol 4) and 6in4 (IP protocol 41) packets to DUT port-1 where
+    *   The outer v4 header has the destination address 203.0.113.1.
+    *   The outer v4 header has the source address ipv4_outer_src_111.
+    *   The outer v4 header has DSCP value has `dscp_encap_no_match` and `dscp_encap_match`
+	*   The outer v4 header has TTL=1
+*  Verify that the punted packets have both ingress_port and
+   target_egress_port metadata set.  target_egress_port should be set
+   to on DUT port-2, port-3, and port-4 per the heirarchical weight.
```

Comment:
Is this target_egress port correct?

As per policy the packet matched with src address ipv4_outer_src_111 with dscp : {dscpEncapA1, dscpEncapA2} and protocol 4 will hit the niDecapTeVrf and in niDecapTeVrf there is no gRIBI action is present (Means no decap/encap rule & no route available)

pfRule3 := &policyFwRule{
		seqID:  3,
		ipv4:   &ipInfo{protocol: 4, dscpSet: []uint8{dscpEncapA1, dscpEncapA2}, sourceAddr: ipv4OuterSrc111WithMask},
		action: &action{decapNI: niDecapTeVrf, postDecapNI: niEncapTeVrfA, decapFallbackNI: niTeVrf111},
	}

Can you please update it according to the vrfSelectionPolicy?